server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java

```diff
@@ -270,4 +274,8 @@ protected long getTraceThreshold() {
     SlowLogLevel getLevel() {
         return level;
     }
+
+    private void setEnabledIfThresholdExceed() {
+        super.setEnabled(this.warnThreshold >= 0 || this.debugThreshold >= 0 || this.infoThreshold >= 0 || this.traceThreshold >= 0);
```

Comment:
This can take the specific thresholdValue as parameter instead of checking all 4 every time

Reply:
And, set enabled only if the input parameter is true

Reply:
Actually, we need to check all 4 values because the setter for one of the thresholds can be invoked during setting update